src/Compilers/CSharp/Portable/Binder/EarlyWellKnownAttributeBinder.cs

```diff
@@ -53,32 +53,10 @@ internal static bool CanBeValidAttributeArgument(ExpressionSyntax node, Binder t
             {
                 // ObjectCreationExpression for primitive types, such as "new int()", are treated as constants and allowed in attribute arguments.
                 case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ImplicitObjectCreationExpression:
```

Comment:
Could you add a test for this case:

```cs
using System.Runtime.InteropServices;

public class C
{
    public void M([DefaultParameterValue(new()) /* also new object() */]object o)
    {
    }
}
```

It looks like the changes would affect the binding of this case. I'm not sure whether it will be _observably_ different, but it should hit this path and potentially bind the object constructor where it wouldn't be before the change.

Reply:
Please also test:

```cs
using System;

[AttributeUsage(new() /* also new AttributeTargets() */)]
public class Attr : Attribute {}
```

This does not compile today. Like the above, I suspect this will continue not working as 0 isn't a valid AttributeTargets value, but it would be good to verify.

Reply:
> It looks like the changes would affect the binding of this case. 

I believe this code doesn't affect regular attribute binding, which all attributes go through. It is used to quickly filter out erroneous attributes applications that could be subject for early binding.

---
In reply to: [566383685](https://github.com/dotnet/roslyn/pull/50862#discussion_r566383685) [](ancestors = 566383685)

Reply:
> I believe this code doesn't affect regular attribute binding, which all attributes go through. It is used to quickly filter out erroneous attributes applications that could be subject for early binding.

`DefaultParameterValueAttribute` is an early-bound attribute, including its parameter values: [http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Source/SourceComplexParameterSymbol.cs,609](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Source/SourceComplexParameterSymbol.cs,609). So this change would cause this helper to return `true` for the filtering code where it did not previously, and potentially cause additional binding.

---
In reply to: [566468839](https://github.com/dotnet/roslyn/pull/50862#discussion_r566468839) [](ancestors = 566468839,566383685)